components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/connector/AdminForcedPasswordResetConfigImplTest.java

```diff
@@ -0,0 +1,149 @@
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_RECOVERY_LINK,
+                "User gets notified with a link to reset password");
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_OTP,
+                "User gets notified with a one time password to try with SSO login");
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_OFFLINE,
+                "An OTP generated and stored in users claims");
+
+        Map<String, String> descriptionMappingExpected = adminForcedPasswordResetConfigIml.getPropertyDescriptionMapping();
+        assertEquals(descriptionMapping, descriptionMappingExpected, "Maps are equal");
+    }
+
+    @Test
+    public void testGetPropertyNames() throws Exception {
+        List<String> properties = new ArrayList<>();
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_RECOVERY_LINK);
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_OTP);
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_OFFLINE);
+
+        String[] propertiesList = properties.toArray(new String[0]);
+        String[] propertiesExpected = adminForcedPasswordResetConfigIml.getPropertyNames();
+
+        for (int i = 0; i < properties.size(); i++) {
+            assertEquals(propertiesList[i], propertiesExpected[i]);
+        }
+    }
+
+    @DataProvider(name = "provideTenantDomain")
+    public Object[][] provideData() {
+        String tenantDomain1;
+        String tenantDomain2;
+
+        tenantDomain1 = "null";
+        tenantDomain2 = "admin";
+        return new Object[][]{
+                {tenantDomain1}, {tenantDomain2}
+        };
+    }
+
+    @Test(dataProvider = "provideTenantDomain")
+    public void testGetDefaultPropertyValues(String tenatDomain) throws Exception {
+        String testEnableAdminPasswordResetWithRecoveryLink = "false";
+        String testEnableAdminPasswordResetWithOTP = "false";
+        String testEnableAdminPasswordResetOffline = "false";
+
+        String testAdminPasswordRecoveryWithLinkProperty = IdentityUtil.getProperty(
+                IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_RECOVERY_LINK);
+        String testAdminPasswordResetWithOTPProperty = IdentityUtil.getProperty(
+                IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_OTP);
+        String testAdminPasswordResetOfflineProperty = IdentityUtil.getProperty(
+                IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_OFFLINE);
+
+        if (StringUtils.isNotEmpty(testAdminPasswordRecoveryWithLinkProperty)) {
+            testEnableAdminPasswordResetWithRecoveryLink = testAdminPasswordRecoveryWithLinkProperty;
+        }
```

Comment:
how abt this instead of if?

assertNotNull(testAdminPasswordRecoveryWithLinkProperty);
testEnableAdminPasswordResetWithRecoveryLink = testAdminPasswordRecoveryWithLinkProperty;

Reply:
Yes. I'll try this.